src/inc/til/size.h

```diff
@@ -0,0 +1,261 @@
+            _width(width),
+            _height(height)
+        {
+
+        }
+
+        template<typename TOther>
+        constexpr size(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().X)> && std::is_integral_v<decltype(std::declval<TOther>().Y)>, int> /*sentinel*/ = 0) :
+            size(static_cast<ptrdiff_t>(other.X), static_cast<ptrdiff_t>(other.Y))
+        {
+
+        }
+
+        template<typename TOther>
+        constexpr size(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().cx)> && std::is_integral_v<decltype(std::declval<TOther>().cy)>, int> /*sentinel*/ = 0) :
+            size(static_cast<ptrdiff_t>(other.cx), static_cast<ptrdiff_t>(other.cy))
+        {
+
+        }
+
+        constexpr bool operator==(const size& other) const noexcept
+        {
+            return _width == other._width &&
+                _height == other._height;
+        }
+
+        constexpr bool operator!=(const size& other) const noexcept
+        {
+            return !(*this == other);
+        }
+
+        size operator+(const size& other) const
+        {
+            ptrdiff_t width;
+            THROW_HR_IF(E_ABORT, !base::CheckAdd(_width, other._width).AssignIfValid(&width));
+
+            ptrdiff_t height;
+            THROW_HR_IF(E_ABORT, !base::CheckAdd(_height, other._height).AssignIfValid(&height));
+
+            return size{ width, height };
+        }
+
+        size operator-(const size& other) const
+        {
+            ptrdiff_t width;
+            THROW_HR_IF(E_ABORT, !base::CheckSub(_width, other._width).AssignIfValid(&width));
+
+            ptrdiff_t height;
+            THROW_HR_IF(E_ABORT, !base::CheckSub(_height, other._height).AssignIfValid(&height));
+
+            return size{ width, height };
+        }
+
+        size operator*(const size& other) const
+        {
+            ptrdiff_t width;
+            THROW_HR_IF(E_ABORT, !base::CheckMul(_width, other._width).AssignIfValid(&width));
+
+            ptrdiff_t height;
+            THROW_HR_IF(E_ABORT, !base::CheckMul(_height, other._height).AssignIfValid(&height));
+
+            return size{ width, height };
+        }
+
+        size operator/(const size& other) const
+        {
+            ptrdiff_t width;
+            THROW_HR_IF(E_ABORT, !base::CheckDiv(_width, other._width).AssignIfValid(&width));
+
+            ptrdiff_t height;
+            THROW_HR_IF(E_ABORT, !base::CheckDiv(_height, other._height).AssignIfValid(&height));
+
+            return size{ width, height};
+        }
+
+        size divide_ceil(const size& other) const
+        {
+            // Divide normally to get the floor.
+            const size floor = *this / other;
+
+            ptrdiff_t adjWidth = 0;
+            ptrdiff_t adjHeight = 0;
+
+            // Check for width remainder, anything not 0.
+            if (_width % other._width)

```

Comment:
May I suggest to use multiplication with `floor._width` for performance reasons? Modulus invokes another slow division that has been already performed to get `floor`.

Reply:
I'm all for performance. Please, do suggest.

But... Sorry, what are you saying? That I should multiply by the already floored number and subtract? Or is there something trickier here that I'm not seeing?

Reply:
You performed division:
`_width / other._width = floor._width`
If you turn it around then should `other._width * floor._width` be equal with `_width` for a division with remainder 0.

So, this should do the trick:
`if (other._width * floor._width != _width)`
For `_height`, respectively.

Reply:
Oh with the comparison. Yes. Yes. OK.